### PR TITLE
Handle the custom desrialization block correctly

### PR DESCRIPTION
**Pull request by sarangan12**
@vishrutshah @salameer Please review

### CODE REVIEW

**Comment by sarangan12** [comment]
@azuresdkci Please retest this

File: runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb [hunk 1]
```diff
@@ -168,8 +168,12 @@ def update_state_from_location_header(request, polling_state, custom_deserializa
     # Updates polling state from Azure async operation header.
     # @param polling_state [MsRestAzure::PollingState] polling state.
     #
-    def update_state_from_azure_async_operation_header(request, polling_state)
-      result = get_async_with_async_operation_deserialization(request)
+    def update_state_from_azure_async_operation_header(request, polling_state, custom_deserialization_block)
+      if custom_deserialization_block.nil?
+        result = get_async_with_async_operation_deserialization(request)
```
Comment (vishrutshah) [inline runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb]: It'd be great if you can combing two methods (get_async_with_async_operation_deserialization & get_async_with_custom_deserialization) into one that can handle `custom_deserialization_block` if provided.

File: runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb [hunk 2]
```diff
@@ -168,8 +168,12 @@ def update_state_from_location_header(request, polling_state, custom_deserializa
     # Updates polling state from Azure async operation header.
     # @param polling_state [MsRestAzure::PollingState] polling state.
```
Comment (vishrutshah) [inline runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb]: please update method documentation based on newly added param.

Comment (sarangan12) [inline runtime/ms_rest_azure/lib/ms_rest_azure/azure_service_client.rb]: Done

**Pull request merged**